src/SignatureHelpProvider.php

```diff
@@ -0,0 +1,211 @@
+<?php
+declare(strict_types = 1);
+
+namespace LanguageServer;
+
+use PhpParser\ErrorHandler\Collecting;
+use PhpParser\Node;
+use LanguageServer\Index\ReadableIndex;
+use LanguageServer\Protocol\{
+    Range,
+    Position,
+    SignatureHelp,
+    SignatureInformation,
+    ParameterInformation
+};
+
+class SignatureHelpProvider
+{
+    /**
+     * @var DefinitionResolver
+     */
+    private $definitionResolver;
+
+    /**
+     * @var ReadableIndex
+     */
+    private $index;
+
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    /**
+     * @var Parser
+     */
+    private $parserErrorHandler;
+
+    /**
+     * @param DefinitionResolver $definitionResolver
+     * @param ReadableIndex      $index
+     */
+    public function __construct(DefinitionResolver $definitionResolver, ReadableIndex $index)
+    {
+        $this->definitionResolver = $definitionResolver;
+        $this->index = $index;
+        $this->parser = new Parser;
+        $this->parserErrorHandler = new Collecting;
+    }
+
+    /**
+     * Returns signature help for a specific cursor position in a document
+     *
+     * @param PhpDocument $doc The opened document
+     * @param Position $pos The cursor position
+     * @return SignatureHelp
+     */
+    public function provideSignature(PhpDocument $doc, Position $pos) : SignatureHelp
+    {
+        $help = new SignatureHelp;
+        $help->signatures = [];
+
+        $handle = fopen('php://temp', 'r+');
+        fwrite($handle, $doc->getContent());
+        fseek($handle, 0);
```

Comment:
We have the whole content as a string, so opening a stream for this provides no benefit at all.
Did you try solving this just with the PhpParser node offsets you get from `getNodeAtPosition()`?